### PR TITLE
Switch from axios-mock-adapter to msw and add api mocks to cypress

### LINKED ISSUE #61: Don't mock axios/fetch for testing api calls, mock server instead with MSW
https://mswjs.io/
https://kentcdodds.com/blog/stop-mocking-fetch

After reading this, this seems like a more robust way to handle api mocks. I was also having many problems with getting `axios-mock-adapter` to work in Cypress (tried for a long time), and in the first few commits in #60 I had to basically have separate mock configurations and strategies for jest and cypress.

MSW does require separate configurations for jest (setupServer) and cypress (setupWorker), but overall is more convenient and invisible, but I'm able to get it working without too much problem ([aside from one oddity discussed here](https://github.com/mswjs/msw/discussions/988)). Network requests behave like normal network requests, and the mocking is fairly seamless. Also it is completely agnostic to how the requests are made; that is, if we ever switch off of `axios`, we don't have to find the new equivalent of `axios-mock-adapter`.

One downside of this is that cypress has a first class `intercept` function that mocks network requests that has nice integration, whereas `axios-mock-adapter` and `msw` is more silent.

Overall, MSW seems like the "Right™" way to do this.

Related reading:
https://github.com/vuejs/vue-test-utils-next/issues/137

### LINKED ISSUE #138: Don't mock axios/fetch for testing api calls, mock server instead with MSW
https://mswjs.io/
https://kentcdodds.com/blog/stop-mocking-fetch

After reading this, this seems like a more robust way to handle api mocks. I was also having many problems with getting `axios-mock-adapter` to work in Cypress (tried for a long time), and in the first few commits in monarch-initiative/monarch-ui-new#60 I had to basically have separate mock configurations and strategies for jest and cypress.

MSW does require separate configurations for jest (setupServer) and cypress (setupWorker), but overall is more convenient and invisible, but I'm able to get it working without too much problem ([aside from one oddity discussed here](https://github.com/mswjs/msw/discussions/988)). Network requests behave like normal network requests, and the mocking is fairly seamless. Also it is completely agnostic to how the requests are made; that is, if we ever switch off of `axios`, we don't have to find the new equivalent of `axios-mock-adapter`.

One downside of this is that cypress has a first class `intercept` function that mocks network requests that has nice integration, whereas `axios-mock-adapter` and `msw` is more silent.

Overall, MSW seems like the "Right™" way to do this.

Related reading:
https://github.com/vuejs/vue-test-utils-next/issues/137

### PR DESCRIPTION
**Pull request by vincerubinetti**
closes monarch-initiative/monarch-api#138 

- have jest ignore inline webpack loaders (these need to be used for other reasons)
- update vue cypress plugin to latest and explicitly install latest cypress
- install `msw` for api call mocking for tests
- abstract out `flushPromises` into own implementation that can be changed later
- fix modal test "not.exist"
- configure `msw` for running as server (for jest/node) and as worker (for cypress/browser)
- specify mocks to `msw`, common to all tests, jest and cypress

### CODE REVIEW

**Comment by vincerubinetti** [comment]
Looking into getting rid of the seemingly unmaintained and unstable `axios-mock-adapter` via this method:
https://stackoverflow.com/questions/51261781/write-test-axios-mock-adapter-with-axios-create

**Review by falquaddoomi — APPROVED** [review]

**Comment by falquaddoomi** [comment]
I read the issue and the articles you suggested, and I don't have anything to add other than that I think it's always better to mock less when you can. MSW, as you said, looks like the right way to do this; I especially like that it runs in both node and the browser, and that it doesn't require modifying the tests too much. I can't weigh in on the Cypress `intercept()` issue either, but it seems like a reasonable sacrifice.

**Comment by vincerubinetti** [comment]
Thanks for taking a look. I'm going to wait until that MSW maintainer responds again to me in that discussion before merging this, just to see if I can get around, or at least _explain_, why we need two `flushPromises()` calls.

I'll poke them again after the holidays.

**Pull request merged**
